### ISSUE TITLE
await FlutterFreshchat.identifyUser() is returning empty string when restoreId is not passed

### ISSUE THREAD

**Issue by daadu** [fayeed/flutter_freshchat#11]
```dart
await FlutterFreshchat.identifyUser(
    externalID: accountId,
    restoreID: null,
  );
```

returns an empty string every time instead of restoreId generated by Freshchat.

PS: I am using the free plan of FreshChat

**Comment by daadu** [comment]
This was tested on Android

**Comment by fayeed** [comment]
@daadu try passing an empty string rather than `null` to `restoreId`

**Comment by daadu** [comment]
Strange. The Activity gets killed or something at times. Anyways, still, the same issue returning an empty string.

**Comment by fayeed** [comment]
@daadu I looked through the issue it doesn't generate a `restoreId` unless you start a conversation after that if you identify the user you should get a `restoreId`. I think this has to with Freshchat, not with this plugin

**Comment by fayeed** [comment]
Also, you can pass `null` to the `identifyUser` method that should work too.

**Comment by daadu** [comment]
Strange! anyways I will implement that way and test.Till then keep this issue open.

**Comment by daadu** [comment]
In my current flow, I identify -> then add user detail -> then start conversation. I guess I should add user detail -> then start conversation -> then identify.

**Comment by fayeed** [comment]
This will also wont work if you are trying to sync conversations so I think the best way would be to check if the user has a `restoreId` stored locally or the server if he has identify the user or if not start the conversation and then identify.

**Comment by fayeed** [comment]
@daadu going to close this issue if the problem persists feel free to open the issue again

**Comment by daadu** [comment]
@fayeed Not working. Sorry for responding late.

Empty string returned by `FlutterFreshchat.identifyUser` even after a conversation is initiated and responded.

**Comment by fayeed** [comment]
I have re-opened this issue again but I don't have time to work on it if you can fix this please do create a pull request or wait till someone else or I fix this.

**Comment by fayeed** [comment]
@daadu I just checked this issue I see no problem, It might be because you also need to update the user info. 

Here I tested it and its working:
![Simulator Screen Shot - iPhone 11 Pro Max - 2019-11-13 at 21 39 17](https://user-images.githubusercontent.com/8464270/68781615-0e9bb780-065e-11ea-8698-a5143a427d41.png)

**Comment by daadu** [comment]
getting an empty string from `identifyUser` for my credentials. My freshchat account is on the free plan (not trial). I am guessing the issue could be related to this.

**Comment by deebhasin** [comment]
I am also facing the same issue...did you find a fix?

**Comment by daadu** [comment]
No, it looks like the issue because of free plan. Although not sure.

**Comment by rahuldange09** [comment]
As per docs. Restore Id is asynchronous and typically only initiated when conversion starts for first time.
So as per their sample app and docs. We have to create Broadcast receiver on android so when it's created for first time. It will be passed by broadcast receiver. (I'm not sure of ios though) 
And then from native android we can call a flutter method which will give restore id when it becomes available.

**Comment by ashim-ux** [comment]
by changing restoreId == "" to restoreId.equals("") in FlutterFreshchatPlugin in android module, empty restoreId issue will be resolved

**Comment by phil10xs** [comment]
> by changing restoreId == "" to restoreId.equals("") in FlutterFreshchatPlugin in android module, empty restoreId issue will be resolved

How about iOS?? I am having issues with the FlutterFreshchat.identifyUser. Prints this error  Could not cast value of type 'NSNull' (0x7fff86b92b38) to 'NSString' (0x7fff86bb7bb0). And crashes when it tries to identify user. What values do I use for externalID  and restoreID to prevent crash..
